WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/ThreatItemBuilderTest.kt

```diff
@@ -0,0 +1,236 @@
+            listOf(UiStringText(FILE_NAME))
+        )
+        // Act
+        val threatItem = buildThreatItem(coreFileModificationThreatModel)
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly for CoreFileModificationThreatModel`() {
+        // Arrange
+        val expectedSubHeader = UiStringRes(R.string.threat_item_sub_header_core_file)
+        // Act
+        val threatItem = buildThreatItem(coreFileModificationThreatModel)
+        // Assert
+        assertThat(threatItem.subHeader).isEqualTo(expectedSubHeader)
+    }
+
+    @Test
+    fun `builds threat header correctly for DatabaseThreatModel`() {
+        // Arrange
+        val expectedHeader = UiStringResWithParams(
+            R.string.threat_item_header_database_threat,
+            listOf(UiStringText("${rows.size}"))
+        )
+        // Act
+        val threatItem = buildThreatItem(databaseThreatModel)
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly for DatabaseThreatModel`() {
+        // Arrange
+        val expectedSubHeader = UiStringText("")
+        // Act
+        val threatItem = buildThreatItem(databaseThreatModel)
+        // Assert
+        assertThat(threatItem.subHeader).isEqualTo(expectedSubHeader)
+    }
+
+    @Test
+    fun `builds threat header correctly for FileThreatModel`() {
+        // Arrange
+        val expectedHeader = UiStringResWithParams(
+            R.string.threat_item_header_file_malicious_code_pattern,
+            listOf(UiStringText(FILE_NAME))
+        )
+        // Act
+        val threatItem = buildThreatItem(fileThreatModel)
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly for FileThreatModel`() {
+        // Arrange
+        val expectedSubHeader = UiStringResWithParams(
+            R.string.threat_item_sub_header_file_signature,
+            listOf(UiStringText(TEST_SIGNATURE))
+        )
+        // Act
+        val threatItem = buildThreatItem(fileThreatModel)
+        // Assert
+        assertThat(threatItem.subHeader).isEqualTo(expectedSubHeader)
+    }
+
+    @Test
+    fun `builds threat header correctly for VulnerableExtensionThreatModel of Plugin type`() {
+        // Arrange
+        val expectedHeader = UiStringResWithParams(
+            R.string.threat_item_header_vulnerable_plugin,
+            listOf(UiStringText(VULNERABLE_THREAT_SLUG), UiStringText(VULNERABLE_THREAT_VERSION))
+        )
+        // Act
+        val threatItem = buildThreatItem(vulnerableExtensionThreatModel)
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly VulnerableExtensionThreatModel of Plugin type`() {
```

Comment:
🔍 Nitpick: Missing "for" -> "correctly for VulnerableExtensionThreatModel"

Reply:
In c5d9fe2